lib/m2ts/ts-correction-stream.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * mux.js
+ *
+ * Copyright (c) 2016 Brightcove
+ * All rights reserved.
+ *
+ * Accepts program elementary stream (PES) data events and corrects
+ * decode and presentation time stamps to account for a rollover
+ * of the 33 bit value.
+ */
+ 
+'use strict';
+
+var Stream = require('../utils/stream');
+
+var MAX_TS = 8589934592;
+
+var RO_THRESH = 4294967296;
+
+var TSCorrectionStream = function() {
+
+  var
+    reference = {
```

Comment:
It depends on where we want the complexity. The current proposal allows us to only have a single `TSCorrectionStream` in the `Transmuxer`. The other option would require the transmuxer to create multiple streams (e.g. 3 different streams in the ts pipeline).